src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -319,6 +319,11 @@ private void SetAnalyzedNullability(BoundExpression expr, VisitResult result, bo
         /// </summary>
         private bool _disableDiagnostics = false;
 
+        /// <summary>
+        /// Whether we are currently visiting as a regular expression or an l-value.
```

Comment:
I find the term "regular expression" a little unclear here. What does it mean when this field is true? That the current expression is an rvalue and not an lvalue?